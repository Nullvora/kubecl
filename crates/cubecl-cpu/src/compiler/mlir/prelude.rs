#![allow(unused)]

pub(super) use super::item::TransformItem;
pub(super) use super::types::TransformType;

pub(super) use melior::Context;
