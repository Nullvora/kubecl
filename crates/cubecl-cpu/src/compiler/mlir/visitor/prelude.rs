pub use super::Visitor;
use tracel_llvm::melior::{Context, ir::Type};

pub trait IntoType {
    fn to_type<'a>(self, context: &'a Context) -> Type<'a>;
    fn is_vectorized(self) -> bool
    where
        Self: Sized,
    {
        false
    }
}
