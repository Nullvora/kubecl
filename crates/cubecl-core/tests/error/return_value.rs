use cubecl_core::prelude::*;

#[cube]
fn range(x: UInt, y: UInt) -> UInt {
    if x == y {
        return x;
    }

    y
}

fn main() {}
