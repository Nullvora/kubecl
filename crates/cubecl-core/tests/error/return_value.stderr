error: Only void return is supported.
 --> tests/error/return_value.rs:7:9
  |
7 |         return x;
  |         ^^^^^^
